sdk/textanalytics/ai-text-analytics/test/textAnalyticsClient.spec.ts

```diff
@@ -258,8 +258,8 @@ describe("[AAD] TextAnalyticsClient", function() {
         assert.fail("Oops, an exception didn't happen.");
       } catch (e) {
         assert.equal(e.statusCode, 400);
-        assert.equal(e.code, "InvalidDocumentBatch");
-        assert.match(e.message, /exceeded the data limitations/);
+        assert.equal(e.code, "InvalidRequest");
```

Comment:
code and message shouldn't change.
Service is returning:
```
{"error":{"code":"InvalidRequest","message":"Invalid document in request.","innererror":{"code":"InvalidDocumentBatch","message":"Batch request contains too many records. Max 5 records are permitted."}}}
```
and we surface the innererror.

Reply:
Our error parsing should be returning the innererror's code, message etc instead of the outer error. This might mean that the error is being correctly parsed. @willmtemple for more context

Reply:
@iscai-msft we get both, you can see the additional checks for the inner ones I added right below this line. Are you suggesting it is not necessary to check the outer ones?

Reply:
Not sure about in JS, but for Python, we really just return the inner error to the customers, the outer error doesn't have any value. From a quick look, it also looks like .net and java are the same as python (@maririos and @mssfang to confirm)

Reply:
@deyaaeldeen just took a dive through js code, and it looks like inner error's should also take precedence in js too: https://github.com/Azure/azure-sdk-for-js/blob/master/sdk/textanalytics/ai-text-analytics/test/error.spec.ts#L26, so I think there's some incorrect parsing going on here

Reply:
Correct on only surfacing innererror

Reply:
@iscai-msft and @maririos I was told that for REST errors like this one that comes from the http pipeline, we throw it as is. We do what you described for errors that come as items in the array we get from the service. This behavior is consistent across JS SDKs. If you are still concerned about this, please talk to @willmtemple as he knows more.

Reply:
This does come from the HTTP layer but might still be breaking in this case. We might need to do something specific for this one & I'll see what we can do about this. It's not all that simple for us to unwrap only one REST error.

Reply:
@iscai-msft and @maririos done.